extensions/positron-python/python_files/posit/positron/converters/pandas_converter.py

```diff
@@ -0,0 +1,242 @@
+        ascending_values = [sk.ascending for sk in self.sort_keys]
+
+        method_chain_setup = [
+            f"column_names = {column_names}",
+            f"asc = {ascending_values}",
+        ]
+
+        method_chain_parts = [".sort_values(by=column_names, ascending=asc)"]
+
+        return method_chain_setup, method_chain_parts
+
+
+class PandasFilterHandler:
+    def __init__(self, filter_key: RowFilter, table_name: str, *, was_series: bool = False):
+        self.filter_key = filter_key
+        self.table_name = table_name
+        self.column_name = "" if was_series else f"[{filter_key.column_schema.column_name!r}]"
+
+    def convert_filters(self) -> Optional[str]:
+        filter_handlers = {
+            RowFilterType.Between: self._convert_between_filter,
+            RowFilterType.NotBetween: self._convert_between_filter,
+            RowFilterType.Compare: self._convert_compare_filter,
+            RowFilterType.IsEmpty: self._convert_is_empty_filter,
+            RowFilterType.NotEmpty: self._convert_not_empty_filter,
+            RowFilterType.IsNull: self._convert_is_null_filter,
+            RowFilterType.NotNull: self._convert_not_null_filter,
+            RowFilterType.IsTrue: self._convert_is_true_filter,
+            RowFilterType.IsFalse: self._convert_is_false_filter,
+            RowFilterType.Search: self._convert_text_search_filter,
+        }
+
+        handler = filter_handlers.get(self.filter_key.filter_type)
+        if handler:
+            return handler()
+
+        if isinstance(self.filter_key.params, FilterSetMembership):
+            return None  # Currently not implemented
+
+        return None
+
+    def _convert_between_filter(self) -> str:
+        assert isinstance(self.filter_key.params, FilterBetween)
+        is_between = self.filter_key.filter_type == RowFilterType.Between
+        left = self.filter_key.params.left_value
+        right = self.filter_key.params.right_value
+        operator = "" if is_between else "~"
+        return f"{operator}{self.table_name}{self.column_name}.between({left}, {right})"
+
+    def _convert_compare_filter(self) -> str:
+        """Handle comparison filters such as equals, not equals, greater than, etc."""
+        assert isinstance(self.filter_key.params, FilterComparison)
+        op = (
+            "=="
+            if self.filter_key.params.op.value == FilterComparisonOp.Eq
+            else self.filter_key.params.op.value
+        )
+
+        value = self.filter_key.params.value
+
+        if self.filter_key.column_schema.type_display == ColumnDisplayType.String:
+            value = repr(value)
+
+        # if it looks date-like, make it a Timestamp, which allows us to handle timezones
+        elif self.filter_key.column_schema.type_display in [
+            ColumnDisplayType.Datetime,
+            ColumnDisplayType.Date,
+        ]:
+            tz = self.filter_key.column_schema.timezone
```

Comment:
if something looks datetime or date-like, we'll always put it in a `pd.Timestamp`, which is able to convert strings.